dynamiqs/rand.py

```diff
@@ -1,32 +1,72 @@
+from __future__ import annotations
+
 import jax
-import numpy as np
 from jax import Array
 from jax import numpy as jnp
+from jaxtyping import PRNGKeyArray
 
 from dynamiqs.utils import dag  # todo: clean this dependency
 
+from .utils.array_types import dtype_complex_to_real, get_cdtype
+
+
+def matrix(
+    shape: tuple[int, ...],
+    *,
+    key: PRNGKeyArray | None = None,
+    dtype: jnp.complex64 | jnp.complex128 | None = None,
+) -> Array:
+    # seed PRNG key if None
+    if key is None:
+        key = jax.random.PRNGKey(42)
```

Comment:
This is IMO a turbo footgun. I removed default argument from my preceding PR because with default people can **think** they are drawing random numbers but they are not until they provide with the key